src/Compilers/CSharp/Test/Emit/CodeGen/IndexAndRangeTests.cs

```diff
@@ -1,12 +1,685 @@
+  IL_0002:  ldc.i4.1
+  IL_0003:  ldc.i4.0
+  IL_0004:  call       ""System.Index..ctor(int, bool)""
+  IL_0009:  ldarg.0
+  IL_000a:  stloc.1
+  IL_000b:  ldloca.s   V_0
+  IL_000d:  call       ""bool System.Index.FromEnd.get""
+  IL_0012:  brtrue.s   IL_001f
+  IL_0014:  ldloc.1
+  IL_0015:  ldloca.s   V_0
+  IL_0017:  call       ""int System.Index.Value.get""
+  IL_001c:  ldelem.i4
+  IL_001d:  br.s       IL_002c
+  IL_001f:  ldloc.1
+  IL_0020:  ldloc.1
+  IL_0021:  ldlen
+  IL_0022:  conv.i4
+  IL_0023:  ldloca.s   V_0
+  IL_0025:  call       ""int System.Index.Value.get""
+  IL_002a:  sub
+  IL_002b:  ldelem.i4
+  IL_002c:  call       ""void System.Console.WriteLine(int)""
+  IL_0031:  ldloca.s   V_0
+  IL_0033:  ldc.i4.1
+  IL_0034:  ldc.i4.1
+  IL_0035:  call       ""System.Index..ctor(int, bool)""
+  IL_003a:  ldarg.0
+  IL_003b:  stloc.1
+  IL_003c:  ldloca.s   V_0
+  IL_003e:  call       ""bool System.Index.FromEnd.get""
+  IL_0043:  brtrue.s   IL_0050
+  IL_0045:  ldloc.1
+  IL_0046:  ldloca.s   V_0
+  IL_0048:  call       ""int System.Index.Value.get""
+  IL_004d:  ldelem.i4
+  IL_004e:  br.s       IL_005d
+  IL_0050:  ldloc.1
+  IL_0051:  ldloc.1
+  IL_0052:  ldlen
+  IL_0053:  conv.i4
+  IL_0054:  ldloca.s   V_0
+  IL_0056:  call       ""int System.Index.Value.get""
+  IL_005b:  sub
+  IL_005c:  ldelem.i4
+  IL_005d:  call       ""void System.Console.WriteLine(int)""
+  IL_0062:  ret
+}");
+        }
+
+        [Fact]
+        public void FakeIndexIndexerArrayNoValue()
+        {
+            var comp = CreateCompilation(@"
+using System;
+namespace System
+{
+    public readonly struct Index
+    {
+        private readonly int _value;
+
+        //public int Value => _value < 0 ? ~_value : _value;
+        public bool FromEnd => _value < 0;
+
+        public Index(int value, bool fromEnd)
+        {
+            if (value < 0)
+            {
+                throw new ArgumentException(""Index must not be negative."", nameof(value));
+            }
+
+            _value = fromEnd? ~value : value;
+        }
+
+        public static implicit operator Index(int value) => new Index(value, fromEnd: false);
+    }
+}
+class C
+{
+    public static void Main()
+    {
+        var x = new[] { 11 };
+        Console.WriteLine(x[^1]);
+        Console.WriteLine(""test""[^1]);
+    }
+}", options: TestOptions.ReleaseExe);
+            comp.VerifyDiagnostics();
+            // We check for the well-known member in lowering, so you won't normally see this
+            // error during binding. This is fine for a preview-only feature.
+            comp.VerifyEmitDiagnostics(
+                // (28,5): error CS0656: Missing compiler required member 'System.Index.Value'
+                //     {
+                Diagnostic(ErrorCode.ERR_MissingPredefinedMember, @"{
+        var x = new[] { 11 };
+        Console.WriteLine(x[^1]);
+        Console.WriteLine(""test""[^1]);
+    }").WithArguments("System.Index", "Value").WithLocation(28, 5));
```

Comment:
Why is the error range the entire method body?

Reply:
This is a result of using the F.WellKnownMember diagnostic handling code. It is recovered in the top of the local rewriter.